common/persistence/persistence_interface.go

```diff
@@ -161,8 +161,8 @@ type (
 		ForkHistoryBranch(ctx context.Context, request *InternalForkHistoryBranchRequest) error
 		// DeleteHistoryBranch removes a branch
 		DeleteHistoryBranch(ctx context.Context, request *InternalDeleteHistoryBranchRequest) error
-		// GetHistoryTree returns all branch information of a tree
-		GetHistoryTree(ctx context.Context, request *InternalGetHistoryTreeRequest) (*InternalGetHistoryTreeResponse, error)
+		// GetHistoryTreeByBranch returns all branch information of a tree
+		GetHistoryTreeByBranch(ctx context.Context, request *InternalGetHistoryTreeByBranchRequest) (*InternalGetHistoryTreeByBranchResponse, error)
```

Comment:
Personally I'd actually vote against renaming this function.  Your new name confused me, and I did most of this PR review thinking your function now returned just the part of the tree that included the Branch, or something like that.  I had to start over :)   

It's not like we have multiple functions that return the tree, some based on the branch, some based on something else.  I'm going to stamp this anyway, I don't feel that strongly about it (though if you DO decide to rename maybe GetHistoryTreeContainingBranch?)

Reply:
Done.